samples/net/sockets/websocket_client/src/main.c

```diff
@@ -100,6 +100,11 @@ static int setup_socket(sa_family_t family, const char *server, int port,
 				LOG_ERR("Failed to set %s secure option (%d)",
 					family_str, -errno);
 				ret = -errno;
+
+				close(*sock);
+				*sock = -1;
+
+				return ret;
```

Comment:
it'suseless, it's done in line 133 already and with this it will miss the test in line 128

Reply:
No if first setsockopt fails it will try to set second setsockopt.

Reply:
have a goto  to line 128 then

Reply:
line 128 tells failed to created socket. In this case socket created succesfully, but failed to do setsockopt.

Reply:
Socket creation, imo, goes with configuring it as well.